### PR TITLE
Scaleway: rename _facts -> _info

### LINKED ISSUE #60520: Scaleway cloud _facts modules need to be renamed to _info and stop returning ansible_facts
##### SUMMARY
According to the [requirements](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#contributing-to-ansible-objective-requirements), `_facts` modules should return information that is specific to the host the module runs on as `ansible_facts`. (See also ansible/ansible#54280 for some discussion.)

I think that the following modules do not satisfy this condition: they return information that is not specific to the host machine, but bound to a cloud account. Please verify whether this is actually the case.

To fix this, the core team thinks the best approach is to rename the `_facts` modules to `_info` modules, and change the return value so that if the module is called with the `_facts` name (should be deprecated, and will then be removed in Ansible 2.13), it should return `ansible_facts` as now (and print a deprecation message), and if it is called with the `_info` name, it no longer returns `ansible_facts`, so that the result needs to be registered. This informs users that they have to change something, won't break their playbooks, and allows them (by switching to the `_info` names) to verify that their roles and playbooks no longer use the facts.

If that approach doesn't work, or you want to change the module's interface (like rename the return value(s)), it is better to deprecate the `_facts` module and create a new `_info` module next to it. To see both approaches implemented, you can take a look at https://github.com/ansible/ansible/commit/2d98734ad51c3de557c3a7d1f7166bdd114ceb0f (ansible/ansible#60178).

If you have any input — preferred method to solve this, doubts that all modules should be renamed, ... — please write it here!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
scaleway_image_facts
scaleway_ip_facts
scaleway_organization_facts
scaleway_security_group_facts
scaleway_server_facts
scaleway_snapshot_facts
scaleway_volume_facts

##### ANSIBLE VERSION
```paste below
2.9.0
```

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #60520.

##### ISSUE TYPE
- Bugfix Pull Request
- New Module Pull Request

##### COMPONENT NAME
scaleway_image_info
scaleway_ip_info
scaleway_organization_info
scaleway_security_group_info
scaleway_server_info
scaleway_snapshot_info
scaleway_volume_info

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

File: lib/ansible/modules/cloud/scaleway/scaleway_image_info.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: scaleway
+
+options:
+
+  region:
+    description:
+    - Scaleway compute zone
```
Comment (samccann) [inline lib/ansible/modules/cloud/scaleway/scaleway_image_info.py]:
```suggestion
      - Scaleway compute zone
```

File: lib/ansible/modules/cloud/scaleway/scaleway_ip_info.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  - Gather information about the Scaleway ips available.
+version_added: "2.9"
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: scaleway
+options:
+  region:
+    description:
+     - Scaleway region to use (for example par1).
```
Comment (samccann) [inline lib/ansible/modules/cloud/scaleway/scaleway_ip_info.py]:
```suggestion
      - Scaleway region to use (for example par1).
```

File: lib/ansible/modules/cloud/scaleway/scaleway_security_group_info.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+description:
+  - Gather information about the Scaleway security groups available.
+version_added: "2.9"
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+options:
+  region:
+    description:
+     - Scaleway region to use (for example par1).
```
Comment (samccann) [inline lib/ansible/modules/cloud/scaleway/scaleway_security_group_info.py]:
```suggestion
      - Scaleway region to use (for example par1).
```

File: lib/ansible/modules/cloud/scaleway/scaleway_server_info.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  - Gather information about the Scaleway servers available.
+version_added: "2.9"
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: scaleway
+options:
+  region:
+    description:
+     - Scaleway region to use (for example par1).
```
Comment (samccann) [inline lib/ansible/modules/cloud/scaleway/scaleway_server_info.py]:
```suggestion
      - Scaleway region to use (for example par1).
```

File: lib/ansible/modules/cloud/scaleway/scaleway_snapshot_info.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  - Gather information about the Scaleway snapshot available.
+version_added: "2.9"
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: scaleway
+options:
+  region:
+    description:
+     - Scaleway region to use (for example par1).
```
Comment (samccann) [inline lib/ansible/modules/cloud/scaleway/scaleway_snapshot_info.py]:
```suggestion
      - Scaleway region to use (for example par1).
```

File: lib/ansible/modules/cloud/scaleway/scaleway_volume_info.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  - Gather information about the Scaleway volumes available.
+version_added: "2.9"
+author:
+  - "Yanis Guenane (@Spredzy)"
+  - "Remy Leone (@sieben)"
+extends_documentation_fragment: scaleway
+options:
+  region:
+    description:
+     - Scaleway region to use (for example par1).
```
Comment (samccann) [inline lib/ansible/modules/cloud/scaleway/scaleway_volume_info.py]:
```suggestion
      - Scaleway region to use (for example par1).
```

File: lib/ansible/modules/cloud/scaleway/scaleway_image_info.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/scaleway/scaleway_image_info.py]: Thanks, updated!

File: lib/ansible/modules/cloud/scaleway/scaleway_ip_info.py (hunk 2 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/scaleway/scaleway_ip_info.py]: Thanks! I also added `C(...)` around `par1`.

**Review by samccann — APPROVED** [review]
Docs portion LGTM

**Comment by felixfontein** [comment]
Rebased to resolve conflict in porting guide.

File: lib/ansible/modules/cloud/scaleway/scaleway_security_group_info.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+'''
+
+EXAMPLES = r'''
+- name: Gather Scaleway security groups information
+  scaleway_security_group_info:
+    region: par1
+  register: result
+
+- debug:
+    msg: "{{ scaleway_security_group_info }}"
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/scaleway/scaleway_security_group_info.py]:
```suggestion
    msg: "{{ result.scaleway_security_group_info }}"
```

**Review by s-hertel — APPROVED** [review]
I checked the diff of each deprecated module with the new _info one, grepped for any stray deprecated module mentions in lib/, test/, and docs/, and manually looked through each modified test since they are unsupported. LGTM.

File: lib/ansible/modules/cloud/scaleway/scaleway_security_group_info.py (hunk 7 above)
Comment (felixfontein) [reply lib/ansible/modules/cloud/scaleway/scaleway_security_group_info.py]: Thanks a lot for catching this!

**Comment by felixfontein** [comment]
@samccann @s-hertel thanks a lot for reviewing!

**Pull request merged**